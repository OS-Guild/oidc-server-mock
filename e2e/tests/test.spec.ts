import axios from "axios";
import { expect } from "chai";
import * as querystring from "querystring";

describe("Test", () => {
    it("should work", async () => {
        const { CLIENT_CREDENTIALS_CLIENT_ID, CLIENT_CREDENTIALS_CLIENT_SECRET, API_RESOURCE } = process.env;
        const params = {
        client_id: CLIENT_CREDENTIALS_CLIENT_ID,
        client_secret: CLIENT_CREDENTIALS_CLIENT_SECRET,
        grant_type: "client_credentials",
        scope: API_RESOURCE,
    };

        const response = await axios.post(process.env.OIDC_TOKEN_URL, querystring.stringify(params));

        // tslint:disable-next-line:no-unused-expression
        expect(response).to.exist;
        // tslint:disable-next-line:no-unused-expression
        expect(response.data.access_token).to.exist;
        // tslint:disable-next-line:no-unused-expression
        expect(response.data.access_token).not.to.be.empty;
    });
});
