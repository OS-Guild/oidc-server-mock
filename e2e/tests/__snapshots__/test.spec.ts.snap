// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test should work 1`] = `
{
  "alg": "RS256",
  "typ": "JWT",
  "iss": "http://localhost:8080",
  "aud": "some-app",
  "client_id": "client-credentials-mock-client",
  "string_claim": "string_claim_value",
  "scope": [
    "some-app-scope-1"
  ],
  "json_claim": [
    "value1",
    "value2"
  ]
}
`;
